### PR TITLE
[flutter_plugin_tool] Fix iOS/macOS naming

### PR DESCRIPTION
**Pull request by stuartmorgan**
Fixes capitalization of iOS and macOS in Dart names to match the Dart
naming rules for two-letter abbreviations, bringing the tool code into
compliance with the recent updates to the Flutter style guide:
https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#capitalize-identifiers-consistent-with-their-spelling

Since this touched two of the `kPlatform...` constants already, I
opportunistically fixed the long-standing naming mistake of those using
a `k` prefix, which isn't done in Dart style.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran [the auto-formatter]. (Unlike the flutter/flutter repo, the flutter/plugins repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. `[shared_preferences]`
- [ ] I listed at least one issue that this PR fixes in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [ ] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

[Contributor Guide]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[the auto-formatter]: https://github.com/flutter/plugins/blob/master/script/tool/README.md#format-code
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

**Comment by gaaclarke** [comment]
Can you add a link to the rule change in the description?

File: script/tool/lib/src/build_examples_command.dart [hunk 1]
```diff
@@ -48,12 +48,12 @@ class BuildExamplesCommand extends PackageLoopingCommand {
...
-    argParser.addFlag(kPlatformMacos);
-    argParser.addFlag(kPlatformWeb);
-    argParser.addFlag(kPlatformWindows);
-    argParser.addFlag(kPlatformWinUwp);
-    argParser.addFlag(kPlatformIos);
+    argParser.addFlag(platformLinux);
+    argParser.addFlag(platformMacOS);
+    argParser.addFlag(platformWeb);
+    argParser.addFlag(platformWindows);
+    argParser.addFlag(platformWinUwp);
```
Comment (gaaclarke) [inline script/tool/lib/src/build_examples_command.dart:55]: Wouldn't you want `platformWinUWP`?

Comment (stuartmorgan) [reply script/tool/lib/src/build_examples_command.dart:55]: UWP is three letters, not two.

Comment (gaaclarke) [reply script/tool/lib/src/build_examples_command.dart:55]: "iOS" is 3 letters too.  It's not a 2 letter acronym.  If this is what you all decided upon that's fine but I think we need to do a better job describing the algorithm like the Java style guide does.  I think the way that is written leaves zero ambiguities.

**Review by gaaclarke — APPROVED** [review]
lgtm as long as thats what the team wants, it seems a bit more ambiguous than the previous rules though imo.

**Comment by Hixie** [comment]
@gaaclarke I added more to @stuartmorgan's style code changes to try to be clearer, PTAL.

File: script/tool/lib/src/build_examples_command.dart (hunk 1 above)
Comment (stuartmorgan) [reply script/tool/lib/src/build_examples_command.dart:55]: UWP is unambiguously not a two-letter acronym (which is what the Dart style guide makes an exception for). `platfromWinUWP` is therefore unambiguously against Dart style.

**Pull request merged**